src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,174 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {

```

Missing test: there is no test that calls `registrationService.register(null)` and asserts that the custom unchecked exception is thrown. The requirements explicitly state: "If User is null, register should throw the custom unchecked exception (implied invalid data path)." Add a test named like `register_nullUser_notOk` that uses `assertThrows(RegistrationException.class, () -> registrationService.register(null));` to cover this required edge case.